packages/lodestar/test/unit/network/gossip/validator.test.ts

```diff
@@ -80,22 +81,47 @@ describe("GossipMessageValidator", () => {
     expect(await validator.isValidIncomingBlock(block)).to.be.equal(false);
   });
 
+  it("should return invalid incoming block - block is too old", async () => {
+    const block = generateEmptySignedBlock();
+    block.message.slot = 10;
+    dbStub.block.isBadBlock.resolves(false);
+    dbStub.block.has.resolves(false);
+    const state = generateState();
+    state.finalizedCheckpoint.epoch = 1000;
+    dbStub.getStateForSlot.resolves(state);
+    expect(await validator.isValidIncomingBlock(block)).to.be.equal(false);
+  });
+
   it("should return invalid incoming block - invalid signature", async () => {
     const block = generateEmptySignedBlock();
     dbStub.block.isBadBlock.resolves(false);
     dbStub.block.has.resolves(false);
     const state = generateState();
-    dbStub.state.getLatest.resolves(state);
+    dbStub.getStateForSlot.resolves(state);
     verifyBlockSignatureStub.returns(false);
     expect(await validator.isValidIncomingBlock(block)).to.be.equal(false);
   });
 
   it("should return valid incoming block", async () => {
     const block = generateEmptySignedBlock();
+    block.message.slot = 10;
     dbStub.block.isBadBlock.resolves(false);
     dbStub.block.has.resolves(false);
     const state = generateState();
-    dbStub.state.getLatest.resolves(state);
+    dbStub.getStateForSlot.resolves(state);
+    verifyBlockSignatureStub.returns(true);
+    expect(await validator.isValidIncomingBlock(block)).to.be.equal(true);
+  });
+
+  it("should return valid incoming block - block in previous epoch", async () => {
```

Comment:
not sure if this helps because we mock everything... that's why I did not create another one for future epoch